pkg/reconciler/revision/reconcile_resources.go

```diff
@@ -249,5 +250,17 @@ func (c *Reconciler) reconcileQueueProxyCertificate(ctx context.Context, rev *v1
 		return fmt.Errorf("certificate in secret %s/%s is not ready yet: private key not found", ns, networking.ServingCertName)
 	}
 
+	// Tell our trackers to reconcile the Revision when our Knative Certificate changes
+	gvk := rev.GetGroupVersionKind()
+	apiVersion, kind := gvk.ToAPIVersionAndKind()
+	if err := c.tracker.TrackReference(tracker.Reference{
+		APIVersion: apiVersion,
+		Kind:       kind,
+		Namespace:  rev.GetNamespace(),
+		Name:       rev.GetName(),
+	}, rev); err != nil {
+		return err
+	}
```

Comment:
You want to track the certificate for changes - but it looks like you're tracking the revision